### ISSUE TITLE
Discover functions in files autoloaded with the autoload.files method in composer.json

### ISSUE THREAD

**Issue by rodrigoaguilera** [phpactor/phpactor#2296]
I use phpactor mainly for Drupal development with neovim and I updated phpactor from `master` of August 2022 to the 2023.4.10 version.
It worked alright even leaving the phpactor default configuration (meaning no special autoloading for Drupal) but after the update I got many errors about `Function "t" not found`.

This `t()` function lives in the `drupal/core` package included via 
```json
{
  "autoload": {
    "files": [
      "includes/bootstrap.inc"
    ]
  }
}
```
https://github.com/drupal/core/blob/10.0.0/composer.json#L106

I played a bit with the configuration of phpactor. First I tried `indexer.include_patterns` with
```
["\/**\/*.php","\/**\/*.inc"]
```
with no success.

Then I tried `"composer.class_maps_only": false` that makes the `t()` function to be included with autocompletion but I don't understand why.

I'm opening this issue as a mix of a bug report because I feel that there might be some regression there as the `t()` function was included back in August with no configuration and also as a support request since I don't understand `composer.class_maps_only` and what problems I might face when I set it to `false`. I'm I doing the right thing dealing with the oddities of Drupal?

**Comment by dantleech** [comment]
I'm not sure what changed, but the indexer has never supported `.inc` files:

https://github.com/phpactor/phpactor/blob/2c9c6dc908f15b3b88fb1db104bce584cedcb711/lib/Indexer/Adapter/Filesystem/FilesystemFileListProvider.php#L31

Changing that to support `.inc` files solves the issue:

![image](https://github.com/phpactor/phpactor/assets/530801/abe999bb-323a-4461-a33b-52eb376cf12a)

We can make the supported PHP file extensions a configuration option I think.

**Comment by rodrigoaguilera** [comment]
I went back to the commit of phpactor that I had from August and you are right, the function was not detected. I guess I'm noticing now because now I get I line prompt in neovim about the function not being found and before it was more silent about it. 

I also did a general upgrade of neovim from 0.7 to 0.9 and all of its plugins.

So the indexer and the composer autoloading work at the same time?
By setting `composer.class_maps_only` to `false` am I including more files to be considered for the language server?
